File: flow-tests/test-scalability/src/test/java/com/vaadin/flow/test/scalability/WarmupFlow.java
package com.vaadin.flow.test.scalability;

import io.gatling.javaapi.core.ScenarioBuilder;
import io.gatling.javaapi.core.Simulation;
import io.gatling.javaapi.http.HttpProtocolBuilder;

import static io.gatling.javaapi.core.CoreDsl.atOnceUsers;
import static io.gatling.javaapi.core.CoreDsl.scenario;
import static io.gatling.javaapi.http.HttpDsl.http;

public class WarmupFlow extends Simulation {

    private HttpProtocolBuilder httpProtocol = http
            .baseUrl("http://localhost:8888").warmUp("http://localhost:8888/")
            .acceptHeader(
                    "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8")
            .acceptEncodingHeader("gzip, deflate")
            .acceptLanguageHeader("en-US,en;q=0.5").userAgentHeader(
                    "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:46.0) Gecko/20100101 Firefox/46.0");

    ScenarioBuilder scn = scenario("Warm up Jetty")
            .exec(http("Open hello world").get("/helloworld/"));

    {
        setUp(scn.injectOpen(atOnceUsers(1000))).protocols(httpProtocol);
    }

}

File: flow-tests/test-scalability/src/test/scala/com/vaadin/flow/test/scalability/WarmupFlow.scala
/*
 * Copyright (C) 2000-2024 Vaadin Ltd
 *
 * This program is available under Vaadin Commercial License and Service Terms.
 *
 * See <https://vaadin.com/commercial-license-and-service-terms> for the full
 * license.
 */
package com.vaadin.flow.test.scalability

import scala.concurrent.duration._

import io.gatling.core.Predef._
import io.gatling.http.Predef._
import io.gatling.jdbc.Predef._

class WarmupFlow extends Simulation {

  val httpProtocol = http
    .baseURL("http://localhost:8888")
    .warmUp("http://localhost:8888/")
    .acceptHeader("text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8")
    .acceptEncodingHeader("gzip, deflate")
    .acceptLanguageHeader("en-US,en;q=0.5")
    .userAgentHeader("Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:46.0) Gecko/20100101 Firefox/46.0")

  var scn = scenario("Warm up Jetty").exec(
    http("Open hello world")
      .get("/helloworld/"))

  setUp(scn.inject(atOnceUsers(1000))).protocols(httpProtocol)
}
